common/app/conf/Filters.scala

```diff
@@ -50,6 +51,20 @@ object BackendHeaderFilter extends Filter with ExecutionContexts {
   }
 }
 
+// See https://www.fastly.com/blog/surrogate-keys-part-1/
+object SurrogateKeyFilter extends Filter with ExecutionContexts {
+
+  private val SurrogateKey = "Surrogate-Key"
+
+  override def apply(nextFilter: (RequestHeader) => Future[Result])(request: RequestHeader): Future[Result] = {
+    val surrogateKey = DigestUtils.md5Hex(request.path)
+    nextFilter(request).map{ result =>
+      val key = result.header.headers.get(SurrogateKey).map(key => s"$key $surrogateKey").getOrElse(surrogateKey)
```

Comment:
My initial PR had a comment about the space - had to redo it when @rich-nguyen decided to upgrade everything. Maybe I will put it back in.